src/Tests/DcsBios/DCSBIOSControlLocatorTests.cs

```diff
@@ -0,0 +1,363 @@
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+            Common.SetEmulationModes(EmulationMode.DCSBIOSInputEnabled);
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var controls = DCSBIOSControlLocator.GetInputControls();
+            foreach (var dcsbiosControl in controls)
+            {
+                var count = dcsbiosControl.Inputs.Count;
+                Assert.True(count > 0);
+
+            }
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5)]
+        public void TestLocatorGetOutputControlsDCSBIOSInputEnabledFlagNotSet(string jsonPath, int dcsAircraftId)
+        {
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
+            if (!Directory.Exists(jsonPath))
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+            // Since Flag DCSBIOSInputEnabled is not set DCSBIOSControlLocator returns null.
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var controls = DCSBIOSControlLocator.GetInputControls();
+            Assert.Null(controls);
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "FLAPS_SWITCH")]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "SASP_MONITOR_TEST")]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "UFC_10")]
+        public void TestLocatorGetLuaCommand(string jsonPath, int dcsAircraftId, string controlId)
+        {
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
+            if (!Directory.Exists(jsonPath))
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+            Common.SetEmulationModes(EmulationMode.DCSBIOSInputEnabled);
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var luaCommand = DCSBIOSControlLocator.GetLuaCommand(controlId, false);
+            Assert.False(string.IsNullOrEmpty(luaCommand));
```

Comment:
Maybe it could be usefull to know if the GetLueAcommand  returns a null or an empty string with an invalid controlId.
 
In a second test like here where we would test the exact value returned. For me a test that says "function should return something" is not very useful. Since the tests is already written, we may as well tests the expected value.

Reply:
Yup, added a test for that.